internal/cmd/update.go

```diff
@@ -0,0 +1,108 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/rhysd/go-github-selfupdate/selfupdate"
+	"github.com/sonatype-nexus-community/nancy/buildversion"
+	"github.com/sonatype-nexus-community/nancy/update"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	rootCmd.AddCommand(updateCmd)
+}
+
+var updateCmd = newUpdateCommand()
+
+func newUpdateCommand() *cobra.Command {
+
+	updateCmd := &cobra.Command{
+		Use: "update",
+		//Short: "Update the tool to the latest version",
+		Short: "Check if there are any updates available",
+		RunE: func(_ *cobra.Command, _ []string) error {
+			//return updateCLI("", true)
+			return updateCLI("", false)
+		},
+	}
+
+	/*	updateCmd.AddCommand(&cobra.Command{
+			Use:    "check",
+			Hidden: true,
+			Short:  "Check if there are any updates available",
+			RunE: func(_ *cobra.Command, _ []string) error {
+				return updateCLI("", false)
+			},
+		})
+	*/
+	/*	updateCmd.AddCommand(&cobra.Command{
+			Use:    "install",
+			Hidden: true,
+			Short:  "Update the tool to the latest version",
+			PersistentPreRun: func(_ *cobra.Command, _ []string) {
+				opts.cfg.SkipUpdateCheck = true
+			},
+			PreRun: func(cmd *cobra.Command, args []string) {
+				opts.args = args
+			},
+			RunE: func(_ *cobra.Command, _ []string) error {
+				return updateCLI(opts)
+			},
+		})
+	*/
```

Comment:
So what are these?? Future state?? If the plan is to have it be `nancy update check` and `nancy update install` we should probably first pass make it actually be `nancy update check` instead of changing the api again probably very soon.

Reply:
> So what are these?? Future state?? If the plan is to have it be `nancy update check` and `nancy update install` we should probably first pass make it actually be `nancy update check` instead of changing the api again probably very soon.

Yeah, far future state. The really nice [CircleCI CLI code](https://github.com/CircleCI-Public/circleci-cli/blob/master/cmd/update.go#L52) I'm basing this off of has some complex (to my brain anyway) logic that sets some flags (`opts...`) on "hidden" commands to affect if/when an "install" is attempted (`dryRun`). I don't think I want to go down the same path. Maybe I'll change my mind if a light bulb goes on for me, but I'm gonna try and keep it simpler.

I kept these commented chunks around mostly to remember to flesh the "install" stuff.

Reply:
Deleted these comments in 1882f98

Created https://github.com/sonatype-nexus-community/nancy/issues/209 to track future work.